src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php

```diff
@@ -115,6 +115,40 @@ public function prepare_item_for_response( $theme, $request ) {
 		$data   = array();
 		$fields = $this->get_fields_for_response( $request );
 
+		$plain_field_mappings = array(
+			'stylesheet' => 'Stylesheet',
+			'template'   => 'Template',
+			'version'    => 'Version',
+		);
+
+		$plain_fields_to_include = array_intersect( array_keys( $plain_field_mappings ), $fields );
+
+		foreach ( $plain_fields_to_include as $field ) {
+			$data[ $field ] = $theme[ $plain_field_mappings[ $field ] ];
+		}
+
+		if ( in_array( 'screenshot', $fields, true ) ) {
+			// Using $theme->get_screenshot() with no args to get absolute URL.
+			$data['screenshot'] = $theme->get_screenshot() ?: '';
+		}
+
+		$rich_field_mappings = array(
+			'author'      => 'Author',
+			'author_uri'  => 'AuthorURI',
+			'description' => 'Description',
+			'name'        => 'Name',
+			'theme_uri'   => 'ThemeURI',
+		);
```

Comment:
Missing fields  `'Tags'  ,'TextDomain' ,DomainPath' ,'RequiresWP' ,'RequiresPHP'`

Reply:
+1 to including these. `tags` should follow the `raw`, `rendered` format.

Reply:
There are also Parent Theme and template fields we should think about as well.

Reply:
> Missing fields `'Tags' ,'TextDomain' ,DomainPath' ,'RequiresWP' ,'RequiresPHP'`

I just added `Tags` (34a179c). I've noticed that the other ones are _header_ fields (i.e. taken from the theme's `style.css` header); however, they're optional and quite 'technical' -- it's conceivable that a REST API client might not be that interested in the required WordPress and PHP versions (if it's e.g. not even programmed in PHP). A similar point can be made for text domain/i18n related fields.

And then, as you mentioned [in your other comment](https://github.com/WordPress/wordpress-develop/pull/222#discussion_r424615493), there's also [these fields](https://github.com/WordPress/wordpress-develop/blob/e18d6e942b7bbc2b776d922d7c9657c4ab1c58b4/src/wp-includes/class-wp-theme.php#L558-L578) (the 'offset' ones, i.e. the ones that can be accessed through square-bracket/pseudo-associative-array syntax), which arguably reflect the public API of the `WP_Theme` class better).

My question is, do we want to reflect
1. the header fields
2. the 'offset' fields
3. the union of 1. and 2.?

I'd personally lean towards option 2.

If this needs more discussion: Would we consider merging this PR pretty much as-is to provide some basic fields, and add the other ones in a follow-up once we've reached consensus?

Reply:
Tags looks good! Let's also add an `items` declaration that specifies each item is a `string` type.

> they're optional and quite 'technical' -- it's conceivable that a REST API client might not be that interested in the required WordPress and PHP versions (if it's e.g. not even programmed in PHP). A similar point can be made for text domain/i18n related fields.

I don't disagree that there are some clients that might not be interested in that data, but there are plenty of fields some clients might not find useful. If they want to restrict the fields returned the fields they need, they can always use the `_fields` parameter. This information is helpful when querying the list of themes in a site. And will be absolutely necessary when we allow switching themes via the REST API.

We are including the text domain in the block types endpoint. Let's leave out domain path for now, since there are alternate ways we could display that and it can't really be utilized.

> And then, as you mentioned in your other comment, there's also these fields 

Most of these we have implemented. The remaining are the parent theme and files. Let's exclude the template/css files for now since those aren't currently public information and will need more auth consideration. For the parent theme, let's omit that until we determine how exactly returning all themes will work.

> which arguably reflect the public API of the WP_Theme class better.

I'm not sure I'd agree with this. I think the named method calls are the preferred way to interact with that data in a meaningful manner. The offsets exist primarily for back-compat with the previous data structure. [Trac Ticket](https://core.trac.wordpress.org/ticket/20103#comment:9).

> If this needs more discussion: Would we consider merging this PR pretty much as-is to provide some basic fields, and add the other ones in a follow-up once we've reached consensus?

My goal will be to commit this this week, if that means we need to punt on some things, we can punt on them, but I think we can probably figure out the changes we need before then.

Reply:
> Tags looks good! Let's also add an `items` declaration that specifies each item is a `string` type.

Thanks, added per 34a179c. BTW I tried searching the REST API docs for an overview of schema props and types but couldn't really find a comprehensive guide. Would you have a pointer for me?

> > they're optional and quite 'technical' -- it's conceivable that a REST API client might not be that interested in the required WordPress and PHP versions (if it's e.g. not even programmed in PHP). A similar point can be made for text domain/i18n related fields.
> 
> I don't disagree that there are some clients that might not be interested in that data, but there are plenty of fields some clients might not find useful. If they want to restrict the fields returned the fields they need, they can always use the `_fields` parameter. This information is helpful when querying the list of themes in a site. And will be absolutely necessary when we allow switching themes via the REST API.

Good point :+1: 

> We are including the text domain in the block types endpoint. Let's leave out domain path for now, since there are alternate ways we could display that and it can't really be utilized.

Thanks! Including `texdomain` per 4a959aee182893a35fbadb78d5330e3e1e48b12d.

> > And then, as you mentioned in your other comment, there's also these fields
> 
> Most of these we have implemented. The remaining are the parent theme and files. Let's exclude the template/css files for now since those aren't currently public information and will need more auth consideration. For the parent theme, let's omit that until we determine how exactly returning all themes will work.

:+1: 

> > which arguably reflect the public API of the WP_Theme class better.
> 
> I'm not sure I'd agree with this. I think the named method calls are the preferred way to interact with that data in a meaningful manner. The offsets exist primarily for back-compat with the previous data structure. [Trac Ticket](https://core.trac.wordpress.org/ticket/20103#comment:9).

Ah, thanks! This has been enlightening. I've taken the opportunity to switch the controller to use `$theme->get()` (which is basically a wrapper around the named methods) rather than array access, d14da4093bc35259e6b74640647378ac8ae1950e.

> > If this needs more discussion: Would we consider merging this PR pretty much as-is to provide some basic fields, and add the other ones in a follow-up once we've reached consensus?
> 
> My goal will be to commit this this week, if that means we need to punt on some things, we can punt on them, but I think we can probably figure out the changes we need before then.

Sounds great!

To recap, I'm reading your comment as agreeing to hold off on adding most fields for now, except for `textdomain` (which I've now added). Please let me know if I was misreading you and if you'd like to see any other fields included!

Reply:
> BTW I tried searching the REST API docs for an overview of schema props and types but couldn't really find a comprehensive guide. Would you have a pointer for me?

We follow JSON Schema version 4, so that'd be the term to search for. The [spec]( https://json-schema.org/specification-links.html#draft-4) is a decent resource. I also wrote an [article](https://timothybjacobs.com/2017/05/17/json-schema-and-the-wp-rest-api/) that I think is decent, ignore the bits about using a separate schema validator.

> Thanks! Including `texdomain` per `4a959ae`.

Looks good!

> I've taken the opportunity to switch the controller to use `$theme->get()`

Looks good!

> Please let me know if I was misreading you and if you'd like to see any other fields included!

I think we should still include the requires PHP and WP headers.

Reply:
> > BTW I tried searching the REST API docs for an overview of schema props and types but couldn't really find a comprehensive guide. Would you have a pointer for me?
> 
> We follow JSON Schema version 4, so that'd be the term to search for. The [spec](https://json-schema.org/specification-links.html#draft-4) is a decent resource. I also wrote an [article](https://timothybjacobs.com/2017/05/17/json-schema-and-the-wp-rest-api/) that I think is decent, ignore the bits about using a separate schema validator.

Thanks for those pointers!

> > Please let me know if I was misreading you and if you'd like to see any other fields included!
> 
> I think we should still include the requires PHP and WP headers.

Done in fac13e5 :+1:

Reply:
lgtm!

Reply:
I _think_ this was the last feedback thread to address. So maybe :crossed_fingers: we're on track for approval? :smile: